tests/sentry/monitors/test_system_incidents.py

```diff
@@ -258,3 +283,261 @@ def test_record_clock_tiock_volume_metric_uniform(metrics, logger):
+def test_tick_decision_anomaly_recovery():
+    start = timezone.now().replace(second=0, microsecond=0)
+
+    test_metrics = [
+        # fmt: off
+        # Operating as normal
+        1.0, 4.0, 3.0, 2.0, 2.0, -3.0,
+        # Anomaly detected
+        -6.0, -7.0,
+        # Anomaly recovers to normal
+        -4.0, -3.0, -3.0, -4.0, -1.0
+        # fmt: on
+    ]
+
+    ts = start
+    fill_historic_metrics(ts, test_metrics)
+
+    # First 6 ticks are operating as normal
+    for _ in range(0, 6):
+        result = update_clock_tick_decisions(ts := ts + timedelta(minutes=1))
+        assert result.decision == TickAnomalyDecision.NORMAL
+        assert result.transition is None
+
+    # Transition into anomalous state (-6)
+    for _ in range(0, 1):
+        result = update_clock_tick_decisions(ts := ts + timedelta(minutes=1))
+        assert result.decision == TickAnomalyDecision.ABNORMAL
+        assert result.transition == AnomalyTransition.ABNORMALITY_STARTED
+
+    # Next 5 ticks (-7, -4, -3, -3, -4) stay in abnormal state
+    for _ in range(0, 5):
+        result = update_clock_tick_decisions(ts := ts + timedelta(minutes=1))
+        assert result.decision == TickAnomalyDecision.ABNORMAL
+        assert result.transition is None
+
+    # Next tick recovers the abnormality after 5 ticks under the abnormality threshold
+    for _ in range(0, 1):
+        result = update_clock_tick_decisions(ts := ts + timedelta(minutes=1))
+        assert result.decision == TickAnomalyDecision.NORMAL
+        assert result.transition == AnomalyTransition.ABNORMALITY_RECOVERED
+
+    # The last 6 ABNORMAL ticks transitioned to NORMAL
+    for i in range(1, 7):
+        assert get_clock_tick_decision(ts - timedelta(minutes=i)) == TickAnomalyDecision.NORMAL
+
+
+@django_db_all
+@override_options({"crons.tick_volume_anomaly_detection": True})
+def test_tick_decisions_simple_incident():
+    """
+    Tests incident detection for an incident that immediately starts and
+    immediately stops.
+    """
+    start = timezone.now().replace(second=0, microsecond=0)
+
+    test_metrics = [
+        # fmt: off
+        # Operating as normal
+        1.0, 4.0, 3.0, 2.0, 2.0, -3.0,
+        # Incident starts immediately
+        -35.0, -80.0, -100.0, -50.0,
+        # Incident quickly recovers
+        -3.0, -2.0, -4.0, -1.0, -4.0
+        # fmt: on
+    ]
+
+    ts = start
+    fill_historic_metrics(ts, test_metrics)
+
+    # First 6 ticks are operating as normal
+    for _ in range(0, 6):
+        result = update_clock_tick_decisions(ts := ts + timedelta(minutes=1))
+        assert result.decision == TickAnomalyDecision.NORMAL
+        assert result.transition is None
+
+    # Transition into incident (-35)
+    for _ in range(0, 1):
+        result = update_clock_tick_decisions(ts := ts + timedelta(minutes=1))
+        assert result.decision == TickAnomalyDecision.INCIDENT
+        assert result.transition == AnomalyTransition.INCIDENT_STARTED
```

Comment:
maybe a little goofy, but I found it easier to read these tests by keeping all the operations inside of these ranges, even if it's only a single iteration